homeassistant/components/awair/__init__.py

```diff
@@ -2,28 +2,42 @@
 from __future__ import annotations
 
 from asyncio import gather
-from typing import Any
 
+from aiohttp.client_exceptions import ClientConnectorError
 from async_timeout import timeout
-from python_awair import Awair
-from python_awair.exceptions import AuthError
+from python_awair import Awair, AwairLocal
+from python_awair.devices import AwairBaseDevice, AwairLocalDevice
+from python_awair.exceptions import AuthError, AwairError
 
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_ACCESS_TOKEN, Platform
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_HOST, Platform
 from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import ConfigEntryAuthFailed
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
-from .const import API_TIMEOUT, DOMAIN, LOGGER, UPDATE_INTERVAL, AwairResult
+from .const import (
+    API_TIMEOUT,
+    DOMAIN,
+    LOGGER,
+    UPDATE_INTERVAL_CLOUD,
+    UPDATE_INTERVAL_LOCAL,
+    AwairResult,
+)
 
 PLATFORMS = [Platform.SENSOR]
 
 
 async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
     """Set up Awair integration from a config entry."""
     session = async_get_clientsession(hass)
-    coordinator = AwairDataUpdateCoordinator(hass, config_entry, session)
+
+    coordinator: AwairDataUpdateCoordinator
+
+    if CONF_HOST in config_entry.data:
+        coordinator = AwairLocalDataUpdateCoordinator(hass, config_entry, session)
+    elif CONF_ACCESS_TOKEN in config_entry.data:
```

Comment:
I guess this can just be `else`
```suggestion
    else:
```